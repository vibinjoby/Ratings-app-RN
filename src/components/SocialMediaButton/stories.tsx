import React from 'react'
import { storiesOf } from '@storybook/react-native'
import SocialMediaButton from './component'
import { StyleSheet } from 'react-native'

storiesOf('Components', module).add('SocialMediaButton', () => (
  <>
    <SocialMediaButton
      mediaText="Continue with Google"
      mediaType="google"
      onPress={() => console.log('pressed')}
      style={styles.btns}
    />
    <SocialMediaButton
      mediaText="Continue with facebook"
      mediaType="fb"
      onPress={() => console.log('pressed')}
      style={styles.btns}
    />
  </>
))

const styles = StyleSheet.create({
  btns: {
    marginVertical: 20,
    marginHorizontal: 20,
  },
})
