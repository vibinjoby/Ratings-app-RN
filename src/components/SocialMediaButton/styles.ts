import { StyleSheet } from 'react-native'

import Colors from '../../utilities/colors'
import Typography from '../../utilities/typography'

export default StyleSheet.create({
  buttonContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderWidth: 1,
    padding: 15,
    backgroundColor: Colors.gBgColor,
    borderColor: Colors.gBgColor,
    borderRadius: 19,
  },
  fbBorder: {
    borderColor: Colors.fbButton,
    backgroundColor: Colors.fbBgColor,
  },
  socialImg: { width: 20, height: 20 },
  buttonText: {
    ...Typography.RegularFont,
    color: Colors.white,
    marginLeft: 20,
    fontSize: 18,
  },
  iconImgFb: {
    width: 10,
    height: 20,
  },
  iconImgGoogle: {
    width: 19,
    height: 19,
  },
  applebtn: {
    width: '100%',
    height: 50,
    marginTop: 20,
  },
})
