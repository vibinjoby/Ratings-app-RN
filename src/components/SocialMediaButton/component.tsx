import React from 'react'
import { TouchableOpacity, Image, Text } from 'react-native'

import styles from './styles'

export interface SocialMediaButtonProps {
  testID?: string
  mediaType: 'fb' | 'google'
  mediaText: string
  style?: Record<string, unknown>
  onPress: () => void
}

const SocialMediaButton: React.FC<SocialMediaButtonProps> = ({
  testID,
  mediaType,
  mediaText,
  style,
  onPress,
}: SocialMediaButtonProps) => {
  return (
    <TouchableOpacity
      testID={testID}
      activeOpacity={0.7}
      onPress={onPress}
      style={[styles.buttonContainer, mediaType === 'fb' && styles.fbBorder, style]}
    >
      <Image
        testID="mediaImg"
        source={
          mediaType === 'fb'
            ? require('../../assets/facebookIcon/facebookIcon.png')
            : require('../../assets/googleIcon/googleIcon.png')
        }
        style={styles.socialImg}
        resizeMode={'contain'}
      />
      <Text style={styles.buttonText}>{mediaText}</Text>
    </TouchableOpacity>
  )
}

export default SocialMediaButton
