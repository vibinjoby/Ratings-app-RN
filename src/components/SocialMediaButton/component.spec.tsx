import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'

import SocialMediaButton from './component'

describe('SocialMediaButton', () => {
  const MOCK_FN = jest.fn()
  const MOCK_MEDIA_TEXT = 'Continue with Google'
  const MOCK_FB_MEDIA_TYPE = 'fb'
  const MOCK_GOOGLE_MEDIA_TYPE = 'google'
  const GOOGLE_URI = require('../../assets/googleIcon/googleIcon.png')
  const FB_URI = require('../../assets/facebookIcon/facebookIcon.png')

  it('should render', () => {
    const { toJSON } = render(
      <SocialMediaButton
        mediaText={MOCK_MEDIA_TEXT}
        mediaType={MOCK_GOOGLE_MEDIA_TYPE}
        onPress={MOCK_FN}
      />,
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should render google button', () => {
    const { getByTestId } = render(
      <SocialMediaButton
        testID="gBtn"
        mediaText={MOCK_MEDIA_TEXT}
        mediaType={MOCK_GOOGLE_MEDIA_TYPE}
        onPress={MOCK_FN}
      />,
    )
    expect(getByTestId('mediaImg').props.source).toBe(GOOGLE_URI)
    fireEvent.press(getByTestId('gBtn'))
    expect(MOCK_FN).toBeCalled()
  })

  it('should render facebook button', () => {
    const { getByTestId } = render(
      <SocialMediaButton
        testID="fbBtn"
        mediaText={MOCK_MEDIA_TEXT}
        mediaType={MOCK_FB_MEDIA_TYPE}
        onPress={MOCK_FN}
      />,
    )
    expect(getByTestId('mediaImg').props.source).toBe(FB_URI)
    fireEvent.press(getByTestId('fbBtn'))
    expect(MOCK_FN).toBeCalled()
  })
})
