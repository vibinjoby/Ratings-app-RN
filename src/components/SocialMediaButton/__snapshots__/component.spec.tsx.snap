// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMediaButton should render 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#5383EC",
      "borderColor": "#5383EC",
      "borderRadius": 19,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "padding": 15,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/assets/googleIcon/googleIcon.png",
      }
    }
    style={
      Object {
        "height": 20,
        "width": 20,
      }
    }
    testID="mediaImg"
  />
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Poppins-Regular",
        "fontSize": 18,
        "marginLeft": 20,
      }
    }
  >
    Continue with Google
  </Text>
</View>
`;
